adobe/table_index.hpp

```diff
@@ -1069,7 +1069,7 @@ inline typename table_index<Key, T, Compare, Transform>::iterator
 table_index<Key, T, Compare, Transform>::lower_bound(const key_type& key) {
     return adobe::lower_bound(
         index_m, key, compare_m,
-        boost::bind(transform_m, bind(indirect<value_type>(), boost::placeholders::_1)));
+        std::bind(transform_m, bind(indirect<value_type>(), std::placeholders::_1)));
```

Comment:
Did you consider turning these into lambdas? I see you did that elsewhere in this PR.